Python/ceval.c

```diff
@@ -2163,9 +2163,17 @@ _PyEval_UnpackIterableStackRef(PyThreadState *tstate, _PyStackRef v_stackref,
             return 1;
         }
         Py_DECREF(w);
-        _PyErr_Format(tstate, PyExc_ValueError,
-                      "too many values to unpack (expected %d)",
-                      argcnt);
+        ll = PySequence_Size(v);
```

Comment:
`PySequence_Size` doesn't support mappings. Perhaps add a call to `PyMapping_Size` as a fallback.

Reply:
You're right, added tests for this as well.